### PR TITLE
issue#71 wishlist changes

### PR DESCRIPTION
**Pull request by Jerry-1947**

### CODE REVIEW

File: frontend/src/components/Sizes.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
     <tr>
       <td>Sizes</td>
       <td>
+        <ul
+          className="sizes-variant"
+          style={{
+            padding: "0px",
+            marginBottom: "10px",
+          }}
+        >
```
Comment (ashdaily) [inline frontend/src/components/Sizes.js:14]: use `mb-*` `p-*` bootstrap classes for consistency. Avoid inline css as much as possible please.

File: frontend/src/context/WishListReducer.js [hunk 2]
```diff
@@ -1,59 +1,65 @@
...
+						localStorage.setItem("wishlistItems", JSON.stringify(wishlistData));
+					} else {
+						localStorage.setItem("wishlistItems", JSON.stringify([]));
+					}
+				}
+			})
+			.catch(() => {
+				localStorage.setItem("wishlistItems", JSON.stringify([]));
+			});
+	}
```
Comment (ashdaily) [inline frontend/src/context/WishListReducer.js:26]: Reminder: don't use tabs, fix this indentation.

File: frontend/src/components/WishlistButton.js [hunk 3]
```diff
@@ -13,24 +16,47 @@ const WishlistButton = ({ data }) => {
...
+      toast.error("Please login first before adding product to wishlist!");
+      return;
+    }
+    axios
+      .post("api/customer/wishlist/", {
+        product: data.id,
+      })
+      .then(() => {
+        addProductToWishlist(data);
+        toast.success("Product added to wishlist!");
```
Comment (ashdaily) [inline frontend/src/components/WishlistButton.js:31]: check status code before toast and hander error as well.

File: frontend/src/components/WishlistButton.js [hunk 4]
```diff
@@ -13,24 +16,47 @@ const WishlistButton = ({ data }) => {
...
+      });
+  };
+
+  const removeFromWishlisht = () => {
+    axios.delete(`api/customer/wishlist/${data.id}/`).then(() => {
+      removeProductFromWishlist(data);
+      toast.success("Product removed to wishlist!");
```
Comment (ashdaily) [inline frontend/src/components/WishlistButton.js:38]: check status code before showing toast and handle error

File: frontend/src/components/WishlistButton.js [hunk 5]
```diff
@@ -13,24 +16,47 @@ const WishlistButton = ({ data }) => {
...
     return (
       <Button
-        variant="danger"
+        variant="outline-danger"
         className="w-100 mt-3"
-        onClick={() => addProductToWishlist(data)}
+        onClick={(data) => addToWishlisht(data)}
       >
-        Add to wishlist <i className="fa fa-heart"></i>
+        Add to wishlist <i className="far fa-heart"></i>
```
Comment (ashdaily) [inline frontend/src/components/WishlistButton.js:59]: there is a typo in this line, ~~far~~ ⛔ , `fa` ✅

File: frontend/src/context/WishListReducer.js (hunk 2 above)
Comment (ashdaily) [inline frontend/src/context/WishListReducer.js:26]: fix the indentation 🤦

File: frontend/src/context/WishListReducer.js [hunk 6]
```diff
@@ -1,59 +1,65 @@
...
 
 export const sumWishlistItems = (items) => {
-  Storage(items);
-  let totalWishlistItems =
-    items.length > 0 ? items.reduce((total) => total + 1, 0) : 0;
-  return { totalWishlistItems };
+	Storage();
+	let totalWishlistItems =
+		items.length > 0 ? items.reduce((total) => total + 1, 0) : 0;
+	return { totalWishlistItems };
```
Comment (ashdaily) [inline frontend/src/context/WishListReducer.js:33]: fix indentation

File: frontend/src/context/WishListReducer.js [hunk 7]
```diff
@@ -1,59 +1,65 @@
...
-          id: action.payload.id,
-        });
-      }
+	switch (action.type) {
+		case "ADD_ITEM_TO_WISHLIST":
+			if (!state.wishlistItems.find((item) => item.id === action.payload.id)) {
+				state.wishlistItems.push({
+					id: action.payload.id,
+				});
+			}
```
Comment (ashdaily) [inline frontend/src/context/WishListReducer.js:43]: indentation

File: frontend/src/context/WishListReducer.js [hunk 8]
```diff
@@ -1,59 +1,65 @@
...
+				),
+				wishlistItems: [
+					...state.wishlistItems.filter(
+						(item) => item.id !== action.payload.id
+					),
+				],
+			};
+		default:
+			return state;
+	}
```
Comment (ashdaily) [inline frontend/src/context/WishListReducer.js:64]: indentation

**Review by ashdaily — CHANGES_REQUESTED** [review]

File: frontend/src/components/WishlistButton.js (hunk 5 above)
Comment (Jerry-1947) [reply frontend/src/components/WishlistButton.js:59]: This is used in font awesome 5

**Review by ashdaily — APPROVED** [review]

**Pull request merged**